iFixit/react-commerce
# Don't report react query retries to Sentry

So I _think_ we were reporting Sentry exceptions even when we hadn't exhausted all available retries in react-query. This changes our api clients to only report to sentry when we've exhausted all retries.

## CR

By default, react query retries 3 times with a backoff of 1 sec, then 2 sec.

React query docs for [`QueryCache`](https://tanstack.com/query/v4/docs/reference/QueryCache#global-callbacks).

## QA

One way to QA this would be to modify the `store/user/cart` api on your subdomain on cominor to return a 400-level response.

Then, set the following env var in `frontend/.env.local`

```
NEXT_PUBLIC_IFIXIT_ORIGIN=https://your-subdomain.cominor.com
```

Then, running localdev, open up any page (like /Parts). Your cart will be fetched for the mini cart, and there should be an error in your console.

The error should also get reported to Sentry.

Closes #827
